spec/models/user_spec.rb

```diff
@@ -0,0 +1,9 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  it { should have_valid(:username).when('Jon', 'Sam')}
+ it { should_not have_valid(:username).when(nil, '')}
+
+ it { should have_valid(:state).when('NH', 'New Hampshire')}
+ it { should_not have_valid(:state).when(nil, '')}
+end
```

Comment:
This should be thoroughly testing everything in the associated user model/table.. Should you be adding tests for Password and Email as well?